consensus/fork.go

```diff
@@ -26,10 +26,14 @@ func (s *State) invalidateNode(node *blockNode) {
 // set of nodes between the common parent and 'bn', starting from the former.
 func (s *State) backtrackToCurrentPath(bn *blockNode) []*blockNode {
 	path := []*blockNode{bn}
-	for s.currentPath[bn.height] != bn.block.ID() {
+	for {
+		// stop when we reach the common parent
+		if bn.height <= s.height() && s.currentPath[bn.height] == bn.block.ID() {
+			break
+		}
```

Comment:
the old `for` loop check was actually masking a `nil` comparison; while `bn.height > s.height()`, the map lookup will return `nil`. Can't pull that trick with slices.